composition-js/src/merging/merge.ts

```diff
@@ -2980,7 +3045,44 @@ class Merger {
         }
       }
     }
+    this.validateContextUsages();
+  }
+
+  // private traverseSelectionSetForType(
+  //   selection: string,
+  //   type: ObjectType,
+  // ) {
+  //   const selectionSet = new SelectionSet(type, )
+  // }
 
+  private validateContextUsages() {
```

Comment:
Is this validation inherently supergraph-level, or could it be expressed by overriding the `validateSubgraphSchema` method of `ContextSpecDefinition` (a handy way to keep validations colocated with the `FeatureDefinition`)?

Reply:
My understanding is this is dead code/needs to be removed, that's been moved to `federation.rs` under the `onValidation()` hook (which is whether we also validate `@requires`).

Reply:
Yeah, I moved this logic into subgraph validation. Deleted dead code.